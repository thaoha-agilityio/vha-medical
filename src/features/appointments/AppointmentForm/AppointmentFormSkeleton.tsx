import { Skeleton } from '@nextui-org/react';

// Types
import { AppointmentModel } from '@/types';

// Components
import { Button, Text } from '@/components/ui';

export type AppointmentModalProps = {
  userId: string;
  role: string;
  data?: AppointmentModel;
  onClose: () => void;
};

export const AppointmentFormSkeleton = ({
  data,
}: {
  data?: AppointmentModel;
}) => {
  const isEdit = !!data;

  return (
    <div className="p-4">
      <Text variant="title" size="xl">
        {isEdit ? 'Update appointment' : 'Create appointment'}
      </Text>

      <div className="flex flex-col md:flex-row md:gap-3 mt-6">
        <div className="w-full text-sm">
          <Text customClass="mb-1">Sender</Text>
          <Skeleton className="w-full h-10 mb-6 rounded-large" />
        </div>
        <div className="w-full text-sm">
          <Text customClass="mb-1">Receiver</Text>
          <Skeleton className="w-full h-10 mb-6 rounded-large" />
        </div>
      </div>

      <div className="flex flex-col md:flex-row md:gap-3 items-start">
        <div className="w-full text-sm">
          <Text customClass="mb-1">Start Date</Text>
          <Skeleton className="w-full h-10 mb-6 rounded-large" />
        </div>
        <div className="w-full text-sm">
          <Text customClass="mb-1">Start Time</Text>
          <Skeleton className="w-full h-10 mb-6 rounded-large" />
        </div>
      </div>

      <div className="w-full text-sm">
        <Text customClass="mb-1">Duration Time</Text>
        <Skeleton className="w-full h-10 mb-6 rounded-large" />
      </div>

      <div className="w-full text-sm">
        <Text customClass="mb-1">Status</Text>
        <Skeleton className="w-full h-10 mb-6 rounded-large" />
      </div>

      <div className="flex gap-4 justify-end mt-3">
        <Button
          variant="outline"
          color="outline"
          className="font-medium"
          isDisabled={true}
        >
          Cancel
        </Button>
        <Button isDisabled={true} type="submit">
          Submit
        </Button>
      </div>
    </div>
  );
};

AppointmentFormSkeleton.displayName = 'AppointmentFormSkeleton';
