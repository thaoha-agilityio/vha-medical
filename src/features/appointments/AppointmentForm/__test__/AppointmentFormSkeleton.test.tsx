import { render } from '@testing-library/react';
import { AppointmentFormSkeleton } from '../AppointmentFormSkeleton';

describe('AppointmentFormSkeleton test case', () => {
  const setup = () => render(<AppointmentFormSkeleton />);

  it('should render correctly', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
