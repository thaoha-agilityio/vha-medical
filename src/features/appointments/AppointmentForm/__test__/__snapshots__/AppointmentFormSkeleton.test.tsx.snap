// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentFormSkeleton test case should render correctly 1`] = `
<DocumentFragment>
  <div
    class="p-4"
  >
    <p
      class="text-primary-100 font-semibold text-xl"
    >
      Create appointment
    </p>
    <div
      class="flex flex-col md:flex-row md:gap-3 mt-6"
    >
      <div
        class="w-full text-sm"
      >
        <p
          class="text-foreground mb-1"
        >
          Sender
        </p>
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 w-full h-10 mb-6 rounded-large"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          />
        </div>
      </div>
      <div
        class="w-full text-sm"
      >
        <p
          class="text-foreground mb-1"
        >
          Receiver
        </p>
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 w-full h-10 mb-6 rounded-large"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          />
        </div>
      </div>
    </div>
    <div
      class="flex flex-col md:flex-row md:gap-3 items-start"
    >
      <div
        class="w-full text-sm"
      >
        <p
          class="text-foreground mb-1"
        >
          Start Date
        </p>
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 w-full h-10 mb-6 rounded-large"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          />
        </div>
      </div>
      <div
        class="w-full text-sm"
      >
        <p
          class="text-foreground mb-1"
        >
          Start Time
        </p>
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 w-full h-10 mb-6 rounded-large"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          />
        </div>
      </div>
    </div>
    <div
      class="w-full text-sm"
    >
      <p
        class="text-foreground mb-1"
      >
        Duration Time
      </p>
      <div
        class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 w-full h-10 mb-6 rounded-large"
      >
        <div
          class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
        />
      </div>
    </div>
    <div
      class="w-full text-sm"
    >
      <p
        class="text-foreground mb-1"
      >
        Status
      </p>
      <div
        class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 w-full h-10 mb-6 rounded-large"
      >
        <div
          class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
        />
      </div>
    </div>
    <div
      class="flex gap-4 justify-end mt-3"
    >
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none border-1 border-secondary-300 hover:bg-linear-success hover:text-content1 text-secondary-300 font-medium"
        data-disabled="true"
        disabled=""
        type="button"
      >
        Cancel
      </button>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent font-semibold bg-linear-success text-content1"
        data-disabled="true"
        disabled=""
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</DocumentFragment>
`;
